sympy/core/relational.py

```diff
@@ -336,7 +336,7 @@ def _eval_simplify(self, **kwargs):
                         else:
                             r = r.func(x, -b / m)
                     else:
-                        r = r.func(b, S.zero)
+                        r = r.func(b, S.Zero)
```

Comment:
Are there tests covering this line?

Reply:
No, as I wrote in the original comment, I cannot figure out how to get there.

Reply:
Basically, this is for a case where `dif` is `m*x + b` where `m.is_zero` is None or True, but `m` should have been evaluated to zero in the subtraction (or previous simplification of arguments or re-evaluation of arguments).

Reply:
Okay, so we need `m` to have no symbols and to be possibly zero but not something that `simplify` can simplify to zero. This can do it:
```python
from sympy import *

x = Symbol('x')
m = exp(1) - exp_polar(1)

simplify(m*x < 1)
```
That gives:
```
Traceback (most recent call last):
  File "t.py", line 6, in <module>
    simplify(m*x < 1)
  File "/home/oscar/current/sympy/sympy.git/sympy/simplify/simplify.py", line 592, in simplify
    return _eval_simplify(**kwargs)
  File "/home/oscar/current/sympy/sympy.git/sympy/core/relational.py", line 339, in _eval_simplify
    r = r.func(b, S.zero)
  File "/home/oscar/current/sympy/sympy.git/sympy/core/singleton.py", line 122, in __getattr__
    name, self))
AttributeError: Attribute 'zero' was not installed on SymPy registry S
```

Reply:
For a test I guess it's better if it's something that `simplify` would never be able to handle e.g.:
```python
from sympy import *

class A(Expr):
    pass

x = Symbol('x')
m = A()

simplify(m*x < 1)
```

Reply:
Great! Thanks! I tried quite a few, but couldn't find anything working.